src/lib/containers/EntityIdList.tsx

```diff
@@ -0,0 +1,42 @@
+import React, { useEffect, useState } from 'react'
+import { useInView } from 'react-intersection-observer'
+import {
+  getEntityTypeByIds
+} from '../utils/SynapseClient'
+
+export type EntityIdListProps = {
+  entityIdList: string[],
+  token: string | undefined
+}
+
+const EntityIdList: React.FC<EntityIdListProps> = props => {
+  const { entityIdList, token } = props
+  const [entityNameList, setEntityNameList] = useState<string>("")
+  const { ref, inView } = useInView()
+  let mounted:boolean = true
+
+  useEffect( () => {
+    if (mounted && inView) {
+      getEntityTypes()
+    }
+    return () => {
+      mounted = false
+    }
+  }, [entityIdList, inView])
+
+  const getEntityTypes = async () => {
+    const entityIds = entityIdList.join(",")
```

Comment:
Prefer this transform (de-structuring) to occur in the getEntityHeadersByIds.  Could you pass the entityIdList array to that function?

Reply:
This is done.